presto-main/src/main/java/com/facebook/presto/operator/Driver.java

```diff
@@ -81,6 +79,12 @@
     @GuardedBy("exclusiveLock")
     private TaskSource currentTaskSource;
 
+    private final Object driverBlockedFutureLock = new Object();
```

Comment:
@martint as you suggested, I updated the `Driver` to set the listeners exactly once (and added a check to `OperatorContext` to verify that). I couldn't set the listeners in the constructor as that would cause unsafe publication of the driver instance. Instead I did it lazily in the `updateDriverBlockedFuture` method (which was named `wakeUpOnRevokeRequest` previously).

This change also required a new lock for the `driverBlockedFuture` as this future will be accessed from multiple threads (threads executing the listeners and the split runner threads).